docs/src/cookbooks/two-side-column.md

```diff
@@ -0,0 +1,892 @@
+
+
+endmodule
+
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule2 extends Module {
+  val in  = IO(Input(UInt(8.W)))
+  val out = IO(Output(UInt(8.W)))
+
+  val differentClock = IO(Input(Clock()))
+  val differentSyncReset = IO(Input(Bool()))
+  
+  val differentAsyncReset = IO(Input(AsyncReset()))
+  
+  
+  
+  val registerWithoutInit = Reg(UInt(8.W))
+  
+  val registerWithInit = RegInit(42.U(8.W))
+  
+  registerWithoutInit := in + 1.U
+  
+  registerWithInit := registerWithInit - 1.U
+  
+  val registerOnDifferentClockAndSyncReset = withClockAndReset(differentClock, differentSyncReset) {
+    val reg = RegInit("h42".U(8.W))
+    reg
+  }
+  registerOnDifferentClockAndSyncReset := in - 1.U
+  
+  val registerOnDifferentClockAndAsyncReset = withClockAndReset(differentClock, differentAsyncReset) {
+    val reg = RegInit("h24".U(8.W))
+    reg
+  }
+  registerOnDifferentClockAndAsyncReset := in + 2.U
+  
+  out := registerWithoutInit + registerWithInit + registerOnDifferentClockAndSyncReset + registerOnDifferentClockAndAsyncReset
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new RegisterModule2)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Case Statements
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module CaseStatementModule(
+  input        clock,
+  input        reset,
+  input  [2:0] a,
+  input  [2:0] b,
+  input  [2:0] c,
+  input  [1:0] sel,
+  output [2:0] out
+);
+  wire [2:0] _GEN_0 = 2'h2 == sel ? c : 3'h0; // @[main.scala 16:16 28:8 14:9]
+  wire [2:0] _GEN_1 = 2'h1 == sel ? b : _GEN_0; // @[main.scala 16:16 24:8]
+  assign out = 2'h0 == sel ? a : _GEN_1; // @[main.scala 16:16 19:8]
+endmodule
+```
+</td>
+
+<td>
+
+```scala mdoc:silent
+class CaseStatementModule extends Module {
+  val a, b, c= IO(Input(UInt(3.W)))
+  val sel = IO(Input(UInt(2.W)))
+  val out = IO(Output(UInt(3.W)))
+  out := 0.U
+
+  switch (sel) {
+    is ("b00".U) {
+      out := a 
+    }
+    is ("b01".U) { 
+      out := b
+    }
+    is ("b10".U) {
+      out := c
+    }
+  }
+};
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementModule)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Case Statements Using Enums 
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module CaseStatementModule(
+  input        clock,
+  input        reset,
+  input  [2:0] a,
+  input  [2:0] b,
+  input        sel,
+  output [2:0] out
+);
+  wire [2:0] _GEN_0 = sel ? b : 3'h0; // @[main.scala 22:16 30:8 20:9]
+  assign out = ~sel ? a : _GEN_0; // @[main.scala 22:16 25:8]
+endmodule
```

Comment:
```suggestion
```

module CaseStatementEnumModule1 (
  input [2:0] rs1,
  input [2:0] pc,
  input AluMux1Sel sel,
  output reg [2:0] out);

  typedef enum {SELECT_RS1, SELECT_PC} AluMux1Sel;

  case(sel)
    SELECT_RS1: out <= rs1;
    SELECT_PC:  out <= pc;
    default:    out <= 3'b0;
  end

endmodule
```

Reply:
Oops this suggested verilog was supposed to go with CaseStatementEnumModule1